src/server/rdb_load.cc

```diff
@@ -516,10 +524,14 @@ void RdbLoaderBase::OpaqueObjLoader::CreateHMap(const LoadTrace* ltrace) {
         return;
       }
     }
-    if (!config_.append) {
-      pv_->InitRobj(OBJ_HASH, kEncodingStrMap2, string_map);
+    if (string_map->Empty() && values_expired) {
+      ec_ = RdbError(errc::value_expired);
```

Comment:
safer to add a return here as if someone will change the code in this function add another step after this if he might miss that you did not return here